crates/matrix-sdk-base/src/store/mod.rs

```diff
@@ -217,15 +226,28 @@ impl Store {
 
     /// Lookup the Room for the given RoomId, or create one, if it didn't exist
     /// yet in the store
-    pub fn get_or_create_room(&self, room_id: &RoomId, room_type: RoomState) -> Room {
+    pub fn get_or_create_room(
+        &self,
+        room_id: &RoomId,
+        room_type: RoomState,
+        client: &BaseClient,
```

Comment:
`self` is always ==  `client`, right? Maybe we can avoid it and use `self` instead?